overlord/ifacestate/helpers.go

```diff
@@ -297,18 +297,19 @@ func CheckInterfaces(st *state.State, snapInfo *snap.Info) error {
 	// XXX: AddImplicitSlots is really a brittle interface
 	snap.AddImplicitSlots(snapInfo)
 
+	if snapInfo.SnapID == "" {
+		// no SnapID means --dangerous was given, so skip interface checks
```

Comment:
@mvo5 dangerous doesn't mean anything when using the store and not a local snap, maybe we should error if it's given in that case (only the local snap case in api.go looks at dangerous)